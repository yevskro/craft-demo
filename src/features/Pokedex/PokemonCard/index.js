import React from 'react';
import styled from 'styled-components';

function PokemonCard({ url, name, width, height }) {
  return (
    <StyledPokemonCard width={width}>
      <img src={url} loading="lazy" alt="" width={width} height={height} />
      <span>{name}</span>
    </StyledPokemonCard>
  );
}

const StyledPokemonCard = styled.div`
  display: inline-block;
  width: ${(props) => props.width};
  text-align: center;
`;

export default PokemonCard;
