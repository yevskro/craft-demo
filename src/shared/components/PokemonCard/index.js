import React from 'react';
import styled from 'styled-components';

function PokemonCard({ url, name, width, height }) {
  return (
    <StyledPokemonCard width={width}>
      <img src={url} loading="lazy" alt="" width={width} height={height} />
      <Name>{name}</Name>
    </StyledPokemonCard>
  );
}

const Name = styled.span`
  text-transform: capitalize;
`;

const StyledPokemonCard = styled.div`
  display: inline-block;
  width: ${(props) => props.width};
  text-align: center;
`;

export default PokemonCard;
