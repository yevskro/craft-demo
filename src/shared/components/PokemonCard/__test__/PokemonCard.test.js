import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PokemonCard from '..';

test('renders without crashing', () => {
  render(<PokemonCard />);
});

test('renders image', () => {
  render(<PokemonCard src="" name="test" alt="bubba" />);
  expect(screen.getByAltText('pokemon card test')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <PokemonCard src="" name="test" alt="bubba" width="96px" height="96px" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
