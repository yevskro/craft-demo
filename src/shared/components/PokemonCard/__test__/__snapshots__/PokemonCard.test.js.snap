// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c1 {
  text-transform: capitalize;
}

.c0 {
  display: inline-block;
  width: 96px;
  text-align: center;
}

<div
  className="c0"
  width="96px"
>
  <img
    alt="pokemon card test"
    height="96px"
    loading="lazy"
    width="96px"
  />
  <span
    className="c1"
  >
    test
  </span>
</div>
`;
